Ah, we can't use const `round` yet because:

```
error[E0015]: cannot call non-const method `<f32 as num_traits::Float>::round` in constant functions
    --> internal/core/graphics/color.rs:62:21
     |
  62 |     (value * 255.0).round() as u8
     |                     ^^^^^^^
     |
note: method `round` is not const because trait `Float` is not const
    --> /github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/num-traits-0.2.19/src/float.rs:932:1
     |
 932 | pub trait Float: Num + Copy + NumCast + PartialOrd + Neg<Output = Self> {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ this trait is not const
...
1196 |     fn round(self) -> Self;
     |     ----------------------- this method is not const
     = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants

```

I'll change the comment.

--- a/internal/compiler/passes/repeater_component.rs
+++ b/internal/compiler/passes/repeater_component.rs
@@ -109,8 +109,7 @@ fn create_repeater_components(component: &Rc<Component>) {
 
                 let enclosing_component =
                     parent_elem.upgrade().unwrap().borrow().enclosing_component.clone();
-                let should_move = Weak::ptr_eq(&enclosing_component, &repeated_component_weak);
-                should_move
+                Weak::ptr_eq(&enclosing_component, &repeated_component_weak)
             }),
         );
 

--- a/internal/compiler/translations.rs
+++ b/internal/compiler/translations.rs
@@ -233,14 +233,14 @@ mod plural_rule_parser {
 
     impl ParsingState<'_> {
         fn skip_whitespace(self) -> Self {
-            let rest = skip_whitespace(self.rest);
+            let rest = self.rest.trim_ascii_start();
             Self { rest, ..self }
         }
     }
 
     /// `<condition> ('?' <expr> : <expr> )?`
     fn parse_expression(string: &[u8]) -> Result<ParsingState<'_>, ParseError<'_>> {
-        let string = skip_whitespace(string);
+        let string = string.trim_ascii_start();
         let state = parse_condition(string)?.skip_whitespace();
         if state.ty != Ty::Boolean {
             return Ok(state);
@@ -258,7 +258,7 @@ mod plural_rule_parser {
                     true_expr: s1.expr.into(),
                     false_expr: s2.expr.into(),
                 },
-                rest: skip_whitespace(s2.rest),
+                rest: s2.rest.trim_ascii_start(),
                 ty: s2.ty,
             })
         } else {
@@ -268,7 +268,7 @@ mod plural_rule_parser {
 
     /// `<and_expr> ("||" <condition>)?`
     fn parse_condition(string: &[u8]) -> Result<ParsingState<'_>, ParseError<'_>> {
-        let string = skip_whitespace(string);
+        let string = string.trim_ascii_start();
         let state = parse_and_expr(string)?.skip_whitespace();
         if state.rest.is_empty() {
             return Ok(state);
@@ -285,7 +285,7 @@ mod plural_rule_parser {
                     op: '|',
                 },
                 ty: Ty::Boolean,
-                rest: skip_whitespace(state2.rest),
+                rest: state2.rest.trim_ascii_start(),
             })
         } else {
             Ok(state)
@@ -294,7 +294,7 @@ mod plural_rule_parser {
 
     /// `<cmp_expr> ("&&" <and_expr>)?`
     fn parse_and_expr(string: &[u8]) -> Result<ParsingState<'_>, ParseError<'_>> {
-        let string = skip_whitespace(string);
+        let string = string.trim_ascii_start();
         let state = parse_cmp_expr(string)?.skip_whitespace();
         if state.rest.is_empty() {
             return Ok(state);
@@ -311,7 +311,7 @@ mod plural_rule_parser {
                     op: '&',
                 },
                 ty: Ty::Boolean,
-                rest: skip_whitespace(state2.rest),
+                rest: state2.rest.trim_ascii_start(),
             })
         } else {
             Ok(state)
@@ -320,9 +320,9 @@ mod plural_rule_parser {
 
     /// `<value> ('=='|'!='|'<'|'>'|'<='|'>=' <cmp_expr>)?`
     fn parse_cmp_expr(string: &[u8]) -> Result<ParsingState<'_>, ParseError<'_>> {
-        let string = skip_whitespace(string);
+        let string = string.trim_ascii_start();
         let mut state = parse_value(string)?;
-        state.rest = skip_whitespace(state.rest);
+        state.rest = state.rest.trim_ascii_start();
         if state.rest.is_empty() {
             return Ok(state);
         }
@@ -346,7 +346,7 @@ mod plural_rule_parser {
                         op,
                     },
                     ty: Ty::Boolean,
-                    rest: skip_whitespace(state2.rest),
+                    rest: state2.rest.trim_ascii_start(),
                 });
             }
         }
@@ -355,9 +355,9 @@ mod plural_rule_parser {
 
     /// `<term> ('%' <term>)?`
     fn parse_value(string: &[u8]) -> Result<ParsingState<'_>, ParseError<'_>> {
-        let string = skip_whitespace(string);
+        let string = string.trim_ascii_start();
         let mut state = parse_term(string)?;
-        state.rest = skip_whitespace(state.rest);
+        state.rest = state.rest.trim_ascii_start();
         if state.rest.is_empty() {
             return Ok(state);
         }
@@ -372,7 +372,7 @@ mod plural_rule_parser {
                     arguments: vec![state.expr, state2.expr],
                 },
                 ty: Ty::Number,
-                rest: skip_whitespace(state2.rest),
+                rest: state2.rest.trim_ascii_start(),
             })
         } else {
             Ok(state)
@@ -380,7 +380,7 @@ mod plural_rule_parser {
     }
 
     fn parse_term(string: &[u8]) -> Result<ParsingState<'_>, ParseError<'_>> {
-        let string = skip_whitespace(string);
+        let string = string.trim_ascii_start();
         let state = match string.first().ok_or(ParseError("unexpected end of string", string))? {
             b'n' => ParsingState {
                 expr: Expression::FunctionParameterReference { index: 0 },
@@ -408,10 +408,6 @@ mod plural_rule_parser {
             .map_err(|_| ParseError("can't parse number", string))?;
         Ok((n, &string[end..]))
     }
-    fn skip_whitespace(string: &[u8]) -> &[u8] {
-        string.trim_ascii_start()
-    }
-
     #[test]
     fn test_parse_rule_expression() {
         #[track_caller]

--- a/internal/core/graphics/color.rs
+++ b/internal/core/graphics/color.rs
@@ -58,7 +58,8 @@ pub struct Color {
     alpha: Channel,
 }
 
-// until slint uses rust 1.90 as MSRV.
+// Can't use f32::round() yet because the Float trait in num_traits isn't const
+// (when building with libm)
 const fn round(mut value: f32) -> u8 {
     if value % 1.0 > 0.5 {
         value += 0.5;

